src/modules/endpoint/sagas/loadCities.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import type { Saga } from 'redux-saga'
+import { createCitiesEndpoint, Payload } from '@integreat-app/integreat-api-client'
+import { call, put } from 'redux-saga/effects'
+import request from '../request'
+import { baseUrl } from '../constants'
+import type { DataContainer } from '../DataContainer'
+import CityModel from '@integreat-app/integreat-api-client/models/CityModel'
+import type { PushCitiesActionType } from '../../app/StoreActionType'
+
+function * loadCities (
+  dataContainer: DataContainer,
+  forceRefresh: boolean
+): Saga<Array<CityModel>> {
+  let cities: Array<CityModel>
+  if (!forceRefresh && (yield call(() => dataContainer.citiesAvailable()))) {
+    cities = yield call(() => dataContainer.getCities())
+  } else {
+    const payload: Payload<Array<CityModel>> = yield call(() => request(createCitiesEndpoint(baseUrl)))
```

Comment:
Maybe we should check if there is a valid internet connection

Reply:
Then the request fails. If that happens, an error will appear.
Once NATIVE-334 is finished, the app will instead of showing an error, use old data to display meaningful content.